src/main/java/com/faforever/api/user/UserService.java

```diff
@@ -124,4 +129,75 @@ void activate(String token) {
     log.debug("User has been activated: {}", user);
     userRepository.save(user);
   }
+
+  void changePassword(String newPassword, User user) {
+    user.setPassword(passwordEncoder.encode(newPassword));
+    userRepository.save(user);
+  }
+
+  void changeLogin(String newLogin, User user) {
+    user.setLogin(newLogin);
```

Comment:
Create an entry in `name_history` table (https://github.com/FAForever/faf-java-api/pull/84)
Optional: You cannot use an name that was used 6 months ago: https://github.com/FAForever/api/pull/163